### PR TITLE
Eng 13180 fix issue2

### PR DESCRIPTION
**Pull request by luyangco**
Serialize all planning/HSQL access. Multi-threaded planning is not supported, so before we can do it all accesses to the planner need to be serialized.

### CODE REVIEW

File: src/frontend/org/voltdb/compiler/PlannerTool.java [hunk 1]
```diff
@@ -197,9 +194,6 @@ public synchronized AdHocPlannedStatement planSql(String sqlIn, StatementPartiti
                 }
             }
 
-            // Reset plan node id counter
-            AbstractPlanNode.resetPlanNodeIds();
```
Comment (nshi) [inline src/frontend/org/voltdb/compiler/PlannerTool.java]: Hmm, this was definitely not thread-safe if another place other the current instance of PlannerTool accesses it.

File: src/frontend/org/voltdb/planner/ParameterizationInfo.java [hunk 2]
```diff
@@ -50,10 +51,10 @@
      * Stores the current count of parameters that we've
      * assigned to the parameter vector in the EE.
      *
-     * TODO: this will need to become thread safe
-     * when parallel planning is possible.
+     * Make it atomic because it's possible to have either both Site thread and
+     * NTService Thread or multiple NTService threads planning in parallel.
      */
-    static private int curParamIndex = 0;
+    static private AtomicInteger curParamIndex = new AtomicInteger(0);
```
Comment (nshi) [inline src/frontend/org/voltdb/planner/ParameterizationInfo.java]: This only prevents it from generating the same ID in different threads, but it doesn't guarantee that the numbers will be monotonically increasing for a given thread. Does that matter? I feel like it does.

Comment (wolffcm) [reply src/frontend/org/voltdb/planner/ParameterizationInfo.java]: Yes, it matters.  We really want the adjacent and starting with zero here.  Also, the numbers we generate need to be deterministic.

File: src/frontend/org/voltdb/compiler/PlannerTool.java (hunk 1 above)
Comment (yzhang1991) [reply src/frontend/org/voltdb/compiler/PlannerTool.java]: Actually, it is also necessary to reset it here, nice catch.

**Pull request merged**